lib/publisher.rb

```diff
@@ -0,0 +1,44 @@
+class Publisher
+  def self.publish(presenter)
+    new(presenter).publish
+  end
+
+  attr_reader :presenter
+
+  def initialize(presenter)
+    @presenter = presenter
+  end
+
+  def publish
+    response = send_item_to_draft_stack
+    response = publish_links if presenter.links && response_successful?(response)
+    response = publish_item if response_successful?(response)
+    response
```

Comment:
Might be better to just return `true` or `false` here then, so the caller doesn't need to know about the internals